R/set_repos.R

```diff
@@ -47,7 +53,13 @@ set_gitlab_repos <- function(gitai,
     GitStats::set_gitlab_host(
       host = host,
       repos = repos,
+      token = get_gitlab_pat(),
```

Comment:
I am not sure about this approach. Basically `GitStats` looks by itself for best-fit access key when `token` parameter is set to `NULL`  (searches through all keys that start with `GITLAB_*`). Now, if you leave `token` parameter empty, tests will pass if you have any key starting with `GITLAB_*` that gives access to public GitLab API. If you will use approach with `get_gitlab_pat()`, test will fail if user does not have explicit `GITLAB_PAT` key as one giving access to public GitLab.

My suggestion would be to create `token` parameter on the level of `set_*_repos()` function which value would be then passed to `GitStats` `set_*_host()` function (here is issue for that: https://github.com/r-world-devs/GitAI/issues/58). By default it would stay `NULL`. For tests we could define it e.g. as `GITLAB_PAT_PUBLIC`. We could still leave it empty and pack it into `withr` the way I do in `GitStats` to test pulling default token:

```
test_that("When empty token for GitLab, GitStats pulls default token", {
  skip_on_cran()
  expect_snapshot(
    withr::with_envvar(new = c("GITLAB_PAT" = Sys.getenv("GITLAB_PAT_PUBLIC")), {
      test_gitstats <- create_gitstats() %>%
        set_gitlab_host(
          orgs = "mbtests"
        )
    })
  )
})
```

Reply:
Good point, I wasn't aware of this functionality of GitStats.
We agreed to add skipping tests directly in the GitStats-related methods.